scheduler/src/cook/mesos/quota.clj

```diff
@@ -96,49 +99,54 @@
    or
    (set-quota! conn \"u1\" :cpus 20.0)
    etc."
-  [conn user reason & kvs]
-  (loop [[type amount & kvs] kvs
-         txns []]
-    (if (nil? amount)
-      @(d/transact conn txns)
-      (if (= type :count)
-        (recur kvs (into [{:db/id (d/tempid :db.part/user)
-                                  :quota/user user
-                                  :quota/count amount}]
-                         txns))
+  [conn user pool-name reason & kvs]
+  (let [pool-name' (pool/pool-name-or-default pool-name)
+        requesting-default-pool? (pool/requesting-default-pool? pool-name)
+        db (d/db conn)]
+    (loop [[type amount & kvs] kvs
+           txns []]
+      (if (nil? amount)
+        @(d/transact conn txns)
         (let [type (resource-type->datomic-resource-type type)
               resource (-> (q '[:find ?r
-                                :in $ ?user ?type
+                                :in $ ?user ?type ?pool-name ?requesting-default-pool
                                 :where
                                 [?e :quota/user ?user]
                                 [?e :quota/resource ?r]
-                                [?r :resource/type ?type]]
-                              (d/db conn) user type)
+                                [?r :resource/type ?type]
+                                [(cook.mesos.pool/check-pool $ ?r :resource/pool ?pool-name
+                                                             ?requesting-default-pool)]]
+                              db user type pool-name' requesting-default-pool?)
                            ffirst)
               txn (if resource
-                    [[:db/add resource :resource/amount amount]]
-                    [{:db/id (d/tempid :db.part/user)
-                      :quota/user user
-                      :quota/resource [{:resource/type type
-                                        :resource/amount amount}]}])]
-          (recur kvs (into txn txns))))))
-  @(d/transact conn [[:db/add [:quota/user user] :quota/reason reason]]))
+                    [[:db/add resource :resource/amount (double amount)]]
+                    (let [resource (cond-> {:resource/type type
+                                            :resource/amount (double amount)}
+                                     pool-name (assoc :resource/pool [:pool/name pool-name]))]
```

Comment:
For whatever reason Cider treats `cond->` as a special case, where arguments on a new line are always indented 2 spaces instead of lining up with the previous arguments (like `let` or `defn`) Do you normally just indent them out to wherever the first argument was? I'm sure I can change the setting.

Reply:
Cursive seems to do this by default, which I guess I've gotten used to:

```clojure
(cond-> {:checkpoint true
         :name (str mesos-framework-name "-" @cook.util/version "-" @cook.util/commit)
         :user ""}
        framework-id (assoc :id {:value framework-id})
        gpu-enabled? (assoc :capabilities [{:type :framework-capability-gpu-resources}])
        mesos-failover-timeout (assoc :failover-timeout mesos-failover-timeout)
        mesos-principal (assoc :principal mesos-principal)
        mesos-role (assoc :role mesos-role))
```

Reply:
Fixed (hopefully)